### PR TITLE
Make wordwrap consistent

### PR DESCRIPTION
**Pull request by gjuric**
As noted in issue #179 the output of word wrap extension depends on the installed **mbstring** extension and if it is installed the `preserve` option is broken.

I am not sure that the output is 100% as intended (there are some edge cases with trimmed spaces), but at least it is consistent with the output of the built in `wordwrap` function for the given examples.

There are also a couple of tests included, but to make sure that the output is the same one has to manually negate the if clause located [here](https://github.com/twigphp/Twig-extensions/blob/bf98d6c99d83620de0938fef76ae17aa71b052a0/lib/Twig/Extensions/Extension/Text.php#L37)  between phpunit runs (or toggle the **mbstring** extension between runs).

**Important!** I adapted [this alghorithm](http://stackoverflow.com/a/4988494) so I am not sure if this has any consequences to the legality of my pull request.

Also, I think pull request #49 should be closed.

### CODE REVIEW

File: lib/Twig/Extensions/Extension/Text.php [hunk 1]
```diff
@@ -55,24 +55,43 @@ function twig_truncate_filter(Twig_Environment $env, $value, $length = 30, $pres
...
     function twig_wordwrap_filter(Twig_Environment $env, $value, $length = 80, $separator = "\n", $preserve = false)
     {
-        $sentences = array();
-
         $previous = mb_regex_encoding();
         mb_regex_encoding($env->getCharset());
 
-        $pieces = mb_split($separator, $value);
-        mb_regex_encoding($previous);
+        $lines = mb_split($separator, $value);
```
Comment (stof) [inline lib/Twig/Extensions/Extension/Text.php]: renaming ``$pieces`` to ``$lines`` is bad, because nothing forces the separator to be a newline char. Please revert this renaming.

File: lib/Twig/Extensions/Extension/Text.php [hunk 2]
```diff
@@ -55,24 +55,43 @@ function twig_truncate_filter(Twig_Environment $env, $value, $length = 30, $pres
...
-                $sentences[] = mb_substr($piece, 0, $length, $env->getCharset());
-                $piece = mb_substr($piece, $length, 2048, $env->getCharset());
+        foreach ($lines as &$line) {
+            $line = rtrim($line);
+            if (mb_strlen($line, $env->getCharset()) <= $length) {
+                continue;
             }
 
-            $sentences[] = $piece;
+            $words = explode(' ', $line);
```
Comment (stof) [inline lib/Twig/Extensions/Extension/Text.php]: you have a reference mismatch here, as you pass the reference variable to a function not taking a reference. Thus, you don't need the reference at all. You could just keep building the ``$sentences`` array instead. Please switch back to this architecture.

Btw, dealing with references makes code harder to understand, as most PHP devs are not familiar with them and with edges cases related to them (were you aware of the performance impact of reference mismatches in PHP 5.x ?)

File: test/Twig/Tests/Extension/TextTest.php [hunk 3]
```diff
@@ -46,4 +46,23 @@ public function getTruncateTestData()
...
+        $output = twig_wordwrap_filter($this->env, $input, $length, $separator, $preserve);
+
+        $this->assertEquals($expectedOutput, $output);
+    }
+
+    public function getWordWrapTestData()
+    {
+        return array(
+            array('This is a very long sentence.', 10, false, '___', 'This is a___very long___sentence.'),
+            array('This is a very long sentence.', 3, false, '___', 'Thi___s___is___a___ver___y___lon___g___sen___ten___ce.'),
```
Comment (stof) [inline test/Twig/Tests/Extension/TextTest.php]: you are missing tests where the separator already appears in the original text

**Comment by fabpot** [comment]
Closing as there is no feedback from the original author and the code cannot be merged as is.
